src/ralph/cli.py

```diff
@@ -45,15 +46,31 @@ class CommaSeparatedKeyValueParamType(click.ParamType):
     def convert(self, value, param, ctx):
         """Split value by comma and equal sign to return a dict build with key/value pairs"""
 
+        # Parse options string
         try:
-            return dict(option.split("=") for option in value.split(","))
+            options = dict(option.split("=") for option in value.split(","))
         except ValueError:
             self.fail(
                 "You should provide key=value pairs separated by commas, e.g. foo=bar,bar=2",
                 param,
                 ctx,
             )
 
+        # Cast simple types
+        for key, value_ in options.items():
+            if value_ == "":
+                options.update({key: None})
+            elif re.match("^(T|t)rue$", value_):
+                options.update({key: True})
+            elif re.match("^(F|f)alse$", value_):
+                options.update({key: False})
+            elif re.match(r"^[0-9]+\.[0-9]+$", value_):
+                options.update({key: float(value_)})
```

Comment:
Are you sure you want to deal with floats ?
Floating point arithmetic and its precision can lead to errors, like in any programming language.
So, to be sure, I would personally avoid dealing with this type and keep it as a string.

Reply:
I hesitated to do so, but in Ralph's particular case, I don't think it's an issue with expected floating point values (1 or 2 digits).